added filter warnings for older versions of python (<3.7)

LucianaMarques opened the pull request:
I simply added the 

if sys.info_version < (3,7):
...code

mentioned in [this issue](https://github.com/ipython/ipython/issues/11333) on the init_deprecation_warnings() functions in the main IPython class. What i understood is that this issue is fixed for python 3.7 but not for older versions, so I guessed this should suffice. 

I wanted opinions from more experienced developers in this project to help to check it and to test it.

LucianaMarques committed:
added filter warnings for older versions of python (<3.7)

This is a "squash" of three commits into one. This means that the intermediate history has been rewritten.

eamanu commented on IPython/core/interactiveshell.py:
```diff
@@ -803,7 +803,10 @@ def init_deprecation_warnings(self):
         """
-        warnings.filterwarnings("default", category=DeprecationWarning, module=self.user_ns.get("__name__"))
+        #warnings.filterwarnings("default", category=DeprecationWarning, module=self.user_ns.get("__name__"))
+        if sys.version_info < 3.7:
```
Change to:

`if sys.version_info < (3, 7)`

This is becasuse sys.version_info on py3 does not support the < operator (or that is my conclussion)

LucianaMarques replied:
Ok! Doing it right now. 

My biggest concern actually was if it resolved what the issue implied. How can I test my code?

Carreau commented:
Thanks !

You did commit all your `.idea` file though. See [here](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore) on how to create a global git ignore file, and add `.idea` to it. 

Then we can walk through how to squash your history for those files to not be added to repo. 

If you code work on Python 3.7 at the REPL you should see the following:
```
In [1]: import warnings
   ...: def foo():
   ...:     warnings.warn('OH Noooo', DeprecationWarning)
   ...:

In [2]: foo()
/Users/bussonniermatthias/anaconda/bin/ipython:3: DeprecationWarning: OH Noooo
  __requires__ = 'ipython'
```

Carreau commented on IPython/core/interactiveshell.py:
```diff
@@ -803,7 +803,9 @@ def init_deprecation_warnings(self):
         """
-        warnings.filterwarnings("default", category=DeprecationWarning, module=self.user_ns.get("__name__"))
+        if sys.version_info < (3,7):
+            warnings.filterwarnings("default", category=DeprecationWarning, module=self.user_ns.get("__name__"))
```
+1 Good job !

Carreau commented:
Good job, and thanks @eamanu for the review !

LucianaMarques commented:
Thank you @Carreau ! I'm new to pull requests, so thank you for sending me this tutorial. 

I'll come back with changes in a bit.

Carreau commented:
Feel free to ask if you want more guidance. I'll stay short on purpose – to gitve you the opportunity to figure it out by yourself –  but I can give you detailed instructions if needed.

LucianaMarques commented:
Thank you @Carreau !
What I did so far:
1) created a global .gitignore_global
2) added ./idea to ir

It wasn't working properly, though, because I had to remove the files from git's tracking. I removed them with git rm -rf .idea and it said that the files inside the folder were succesfully removed. But whenever I tried to commit and push, the repository was already up-to-date.

3) so I decided to exclude the ./idea folder

and it worked. 

Does this sound right? Is there a better way to do it?

Thanks

Carreau commented:
It's good, let me show you how to rewrite the history to have only one clean commit and not have the intermediate .idea files in history.

We'll rebase interactive (`-i`) and squash (`s`) the last two commits


```
$ git rebase -i HEAD~3
<vim open if you haven't set $EDITOR>
<press i and Edit the second and third line, replace "pick" by "s">
<press ESC : w q Enter>
< write your new commit message>
```

Now you can force push. 

Here is it as a gif.

![rebase example](https://user-images.githubusercontent.com/335567/46921876-c1aa6500-cfb5-11e8-99df-0055bcb83b03.gif)


If it's too complicated I can do that for you.

LucianaMarques commented:
Hi there @Carreau ! 

Thank you for the help, I believe I did successfully what you asked for. This GIF is great!

What are the next steps now? should we do some more testing?

LucianaMarques commented:
> Great!

Than you @eamanu !

Carreau commented:
> What are the next steps now? should we do some more testing?

No more step for this issue that you. For this specific issue I don't think more testing is necessary I'll just update 11333 so that once we drop 3.6 the code can actually be removed.... but that will take a couple of years. 

Good job!

LucianaMarques commented:
> Good job!

@Carreau  thank you! Now I'm looking for more issues to work on, do you have any suggestions?

Carreau commented:
> do you have any suggestions?

Not particularly, I'll have to go through and figure out if issues are easy or hard. 

You can try to go through the list of what is needed for [`7.1 release`](https://github.com/ipython/ipython/milestone/59) some of them have been fixed and just need to be closed. 

You should be able to also apply a [subset of labels to issues](https://github.com/ipython/ipython/blob/master/.meeseeksdev.yml#L11-L17) by gently asking our bot like so: 

@meeseeksdev tag hacktoberfest

I'll try to mark a few issue as "good first issue" and/or "help wanted"
